bgpd/bgp_evpn.c

```diff
@@ -7086,16 +7058,33 @@ int bgp_evpn_local_l3vni_add(vni_t l3vni, vrf_id_t vrf_id,
 	/* PIP user configured mac is not present use svi mac as sys mac */
 	if (is_zero_mac(&bgp_vrf->evpn_info->pip_rmac_static))
 		memcpy(&bgp_vrf->evpn_info->pip_rmac, svi_rmac, ETH_ALEN);
+	/* for v6 vtep_ip assign lo primary v6 address as pip,
+	 * for v4 vtep_ip bgp instance router-id as pip in bgp_evpn_init.
+	 */
+	if (IS_IPADDR_V6(&bgp_vrf->originator_ip)) {
+		struct interface *ifp;
+		struct in6_addr addr;
+
+		ifp = if_get_vrf_loopback(VRF_DEFAULT);
```

Comment:
is VRF_DEFAULT correct - didn't we add non-default-vrf support in more paths?

Reply:
so advertise-pip uses bgp default instance router-id hence, default instance is the ideal for this purpose.